relnotes/stringz.deprecation.md

```diff
@@ -0,0 +1,5 @@
+* `ocean.stdc.stringz`
+
+  This module and all functions inside of it have been deprecated,
+  as they duplicates already existing functionalities in ocean.text.util.StringC,
+  which is more idiomatic and actively maintained.
```

Comment:
Maybe add a comment here suggesting to switch to `static mstring buffer` when necessary? I would expect more questions like @gavin-norman-sociomantic had would arise.